pkg/security/security_profile/dump/activity_dump.go

```diff
@@ -1165,26 +1165,37 @@ func (pan *ProcessActivityNode) debug(w io.Writer, prefix string) {
 
 // scrubAndReleaseArgsEnvs scrubs the process args and envs, and then releases them
 func (pan *ProcessActivityNode) scrubAndReleaseArgsEnvs(resolver *sprocess.Resolver) {
-	_, _ = resolver.GetProcessScrubbedArgv(&pan.Process)
-	envs, envsTruncated := resolver.GetProcessEnvs(&pan.Process)
-	pan.Process.Envs = envs
-	pan.Process.EnvsTruncated = envsTruncated
-	pan.Process.Argv0, _ = resolver.GetProcessArgv0(&pan.Process)
+	if pan.Process.ArgsEntry != nil {
+		_, _ = resolver.GetProcessScrubbedArgv(&pan.Process)
+		pan.Process.Argv0, _ = resolver.GetProcessArgv0(&pan.Process)
+		pan.Process.ArgsEntry = nil
 
-	pan.Process.ArgsEntry = nil
-	pan.Process.EnvsEntry = nil
+	}
+	if pan.Process.EnvsEntry != nil {
+		envs, envsTruncated := resolver.GetProcessEnvs(&pan.Process)
+		pan.Process.Envs = envs
+		pan.Process.EnvsTruncated = envsTruncated
+		pan.Process.EnvsEntry = nil
+	}
 }
 
 // Matches return true if the process fields used to generate the dump are identical with the provided ProcessCacheEntry
-func (pan *ProcessActivityNode) Matches(entry *model.ProcessCacheEntry, matchArgs bool, processResolver *sprocess.Resolver) bool {
-
+func (pan *ProcessActivityNode) Matches(entry *model.Process, matchArgs bool) bool {
 	if pan.Process.Comm == entry.Comm && pan.Process.FileEvent.PathnameStr == entry.FileEvent.PathnameStr &&
 		pan.Process.Credentials == entry.Credentials {
 
 		if matchArgs {
-
-			panArgs, _ := processResolver.GetProcessArgv(&pan.Process)
-			entryArgs, _ := processResolver.GetProcessArgv(&entry.Process)
+			var panArgs, entryArgs []string
+			if pan.Process.ArgsEntry != nil {
```

Comment:
Not specially related to this PR directly but I think we should handle all this caching "ArgsEntry" thing in the Process context itself

Reply:
Discussed and yep for sure, this part needs a rework and should be done on another PR